### PR TITLE
Synthetic A records for autoscaling groups

### PR DESCRIPTION
**Pull request by steveniemitz**

### CODE REVIEW

File: src/dlz_aws.cpp [hunk 1]
```diff
@@ -196,6 +213,22 @@ isc_result_t dlz_lookup(
     }
   }
 
+  if (StringUtils::CaselessCompare(state->autoscaler_zone_name.c_str(), zone)) {
+    std::string clientAddr;
+    std::vector<std::string> nodes;
+    get_src_address(methods, clientinfo, &clientAddr);
+    if (state->client->TryResolveAutoscaler(name, clientAddr, &nodes)) {
```
Comment (GeorgeSirois) [inline src/dlz_aws.cpp:219]: Is "name" everything under the zone, or just the first level?

e.g. if I want to set the dns-alias tag to "general.kafka" on the ASG, will "general.kafka.asg.aws.tcdc.io" resolve correctly?

**Comment by GeorgeSirois** [comment]
Also "Syntheic" -> "Synthetic" in the commit message.

File: src/dlz_aws.cpp (hunk 1 above)
Comment (steveniemitz) [inline src/dlz_aws.cpp:219]: Yeah, it should.  The magic of DNS will try both general.[kafka.asg] and general.kafka.[asg]

File: src/dlz_aws.cpp [hunk 2]
```diff
@@ -196,10 +212,25 @@ isc_result_t dlz_lookup(
...
+      size_t maxNodes = std::min(nodes.size(), state->num_asg_records);
+      for (const auto&& node : k_random<std::string>(nodes, maxNodes)) {
+        state->callbacks.putrr(lookup, "A", 120, node.c_str());
```
Comment (GeorgeSirois) [inline src/dlz_aws.cpp:222]: We might want to make the timeout configurable (or hijack instance timeout), but we can save that for another CL.

**Comment by GeorgeSirois** [comment]
:zzz: ap

File: src/dlz_aws.cpp (hunk 2 above)
Comment (steveniemitz) [inline src/dlz_aws.cpp:222]: yeah, agreed.

**Pull request merged**